src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -102,8 +111,14 @@ def self.getModules(env)
       # NOTE: If in the future we want to collect more
       #       Module settings, this should be more programatic
       #       rather than getting these settings one by one
-      module_data[:name] = mod.forge_name
-      module_data[:version] = mod.version
+      if mod.forge_name
+        module_data[:name] = mod.forge_name
+        module_data[:version] = mod.version
```

Comment:
@camlow325 - yeah, so far as I can tell if one of [these](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/module.rb#L165) keys are missing from the metadata.json file, you can't operate at all on the modules object. So the entire iteration gets skipped and we can't fail over to a sensible default like empty string. It's definetly possible that someone could have a metadata.json file and have some of those keys missing, but I'm not sure how likely it is or what else will also be broken if they try to use those modules.